src/modules/interface/param.h

```diff
@@ -172,16 +172,20 @@ struct param_s {
 #ifndef UNIT_TEST_MODE
 
 #define PARAM_ADD(TYPE, NAME, ADDRESS) \
-   { .type = TYPE, .name = #NAME, .address = (void*)(ADDRESS), },
+  { .type = TYPE, .name = #NAME, .address = (void*)(ADDRESS), },
+
+// The callback notification function will run from the param task, it should not block and should run quickly.
+#define PARAM_ADD_WITH_CALLBACK(TYPE, NAME, ADDRESS, CALLBACK) \
+  { .type = TYPE | PARAM_CALLBACK, .name = #NAME, .address = (void*)(ADDRESS), .callback = (void *)CALLBACK, },
```

Comment:
This will cause problems everywhere we check type to see what type this is... right? And in the elf-sanity script that also checks types.

Reply:
For instance in param.c `paramWriteByNameProces`:

```c
  if (type != (params[ptr].type & (~(PARAM_CORE | PARAM_RONLY)))) {
    return EINVAL;
  }
  ```

This will break by this change?

Reply:
Possibly yes but I thought it wouldn't as the PARAM_CALLBACK is using an unused bit. I have to investigate.